### PR TITLE
Feat/prettier blog pagination routing

### PR DESCRIPTION
**Pull request by Frankwarnaar**
## What changes were made

- Use the naked /blog route for page 1, while using blog/page/x for other page numbers
- Refactor `pagination-nav` to be more context independent. It now should be useable for any page, no matter the route structure.

## How to test or check results

- Check the blog overview routes
- Navigating to /blog/page/1 should result in a redirect to the naked url

## Checks
- [x] All content model changes are done through [scripted migrations](../readme.md#scripted-migrations)
- [x] Appointed PR reviewers

### CODE REVIEW

File: src/components/pagination-nav/pagination-nav.vue [hunk 1]
```diff
@@ -56,27 +56,19 @@
...
-      type: String,
-      default: null,
-    },
-  })
+  const props = defineProps<{
+    totalItems: number
+    currentPage: number
+    perPage: number
+    // eslint-disable-next-line no-unused-vars
+    getPaginatedRoute: (page: number) => number | RouteLocation
```
Comment (lsarni) [inline src/components/pagination-nav/pagination-nav.vue:70]: When is the return a number?

**Review by lsarni — APPROVED** [review]

File: src/components/pagination-nav/pagination-nav.vue (hunk 1 above)
Comment (Frankwarnaar) [reply src/components/pagination-nav/pagination-nav.vue:70]: Never. Good catch! :)

**Comment by Frankwarnaar** [comment]
Hmm, somehow the redirect on /blog/page/1 happens client side... I want it server side

**Comment by Frankwarnaar** [comment]
> Hmm, somehow the redirect on /blog/page/1 happens client side... I want it server side

Ah the redirect was configured in DatoCMS. I changed it and now it works like a charm 😍

**Pull request merged**